src/storage/seg/src/item/mod.rs

```diff
@@ -83,6 +89,99 @@ impl std::fmt::Debug for Item {
     }
 }
 
+/// Items are the base unit of data stored within the cache.
+pub struct RichItem {
```

Comment:
Is there a reason this can't be:
```pub struct RichItem {
  item: Item,
  item_info: u64,
  item_info_ptr: *const u64,
}```

Reply:
nope, your proposal is better, I can change it :)